static/js/containers/ProfileImage.js

```diff
@@ -30,43 +30,52 @@ const formatPhotoName = photo => (
 
 class ProfileImage extends React.Component {
   props: {
-    profile:              Profile,
+    afterImageUpload:     ?(o: ProfileFetchResponse) => void,
+    clearPhotoEdit:       () => void,
+    dispatch:             Dispatch,
     editable:             boolean,
     imageUpload:          Object,
-    dispatch:             Dispatch,
-    clearPhotoEdit:       () => void,
-    setDialogVisibility:  (b: boolean) => void,
-    updatePhotoEdit:      (b: Blob) => void,
-    startPhotoEdit:       (p: File) => void,
+    linkText:             string,
     photoDialogOpen:      boolean,
+    profile:              Profile,
+    setDialogVisibility:  (b: boolean) => void,
     setPhotoError:        (s: string) => void,
+    showLink:             boolean,
+    startPhotoEdit:       (p: File) => void,
+    updatePhotoEdit:      (b: Blob) => void,
   };
 
   static defaultProps = {
     editable: false
   };
 
-  updateUserPhoto: Function = () => {
+  updateUserPhoto = () => {
     const {
       profile: { username },
       imageUpload: { edit, photo },
       dispatch,
       clearPhotoEdit,
       setDialogVisibility,
+      afterImageUpload,
     } = this.props;
+
     return dispatch(updateUserPhoto(username, edit, formatPhotoName(photo))).then(() => {
       clearPhotoEdit();
       setDialogVisibility(false);
-      this.fetchUserProfile();
+      return this.fetchUserProfile().then(arg => {
```

Comment:
well, we only care about the first one. I meant to rename this to `resp` or similar.